wallet/src/main/scala/org/bitcoins/wallet/internal/RescanHandling.scala

```diff
@@ -0,0 +1,157 @@
+package org.bitcoins.wallet.internal
+
+import java.util.concurrent.Executors
+
+import org.bitcoins.core.crypto.DoubleSha256Digest
+import org.bitcoins.core.hd.HDChainType
+import org.bitcoins.core.protocol.script.ScriptPubKey
+import org.bitcoins.core.protocol.{BitcoinAddress, BlockStamp}
+import org.bitcoins.core.util.FutureUtil
+import org.bitcoins.wallet.models.AddressDb
+import org.bitcoins.wallet.{LockedWallet, WalletLogger}
+
+import scala.annotation.tailrec
+import scala.concurrent.{ExecutionContext, Future}
+
+private[wallet] trait RescanHandling extends WalletLogger {
+  self: LockedWallet =>
+
+  val AddressGapLimit = 20
+
+  /////////////////////
+  // Public facing API
+
+  /** @inheritdoc */
+  override def rescanNeutrinoWallet(
+      startOpt: Option[BlockStamp],
+      endOpt: Option[BlockStamp],
+      addressBatchSize: Int): Future[Unit] = {
+
+    logger.info(s"Starting rescanning the wallet.")
+
+    val res = for {
+      _ <- spendingInfoDAO.deleteAll()
+      _ <- addressDAO.deleteAll()
```

Comment:
I'm not sure if I agree with deleting everything? It seems like it is a bad idea to destroy data that already exists in the wallet, is there a reason we can't just update/create?